integrationtests/src/test/java/io/kroxylicious/proxy/NettyLeakLogAppender.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.Layout;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.Property;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+import org.junit.jupiter.api.Assertions;
+
+import io.netty.util.ResourceLeakDetector;
+
+@Plugin(name = "NettyLeakLogAppender", category = "Core", elementType = "appender", printObject = true)
+public class NettyLeakLogAppender extends AbstractAppender {
```

Comment:
Thinking out loud I wonder if its worth adding this to the junit extension so we can apply it to all the integration tests?

Its the sort of thing I think would be good to support filter authors in verifying as well.

Reply:
is the junit5 extension that right please?  the extension doesn't actually know anything about netty.

Reply:
Yeah I did wonder about that. Maybe a separate extension in Kroxylicious test tools